packages/react-components/src/components/MessageThread.tsx

```diff
@@ -909,27 +936,44 @@ export const MessageThread = (props: MessageThreadProps): JSX.Element => {
             onRenderAvatar={onRenderAvatar}
             showMessageStatus={showMessageStatus}
             messageStatus={messageProps.message.status}
+            onActionButtonClick={onActionButtonClickMemo}
           />
         );
       }
       return <></>;
     },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [new Date().toDateString(), isNarrow, messageThreadParticipantCount, onRenderAvatar]
+    [
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+      new Date().toDateString(),
+      isNarrow,
+      messageThreadParticipantCount,
+      onRenderAvatar,
+      onActionButtonClickMemo,
+      props.onRenderFileDownloads,
+      props.userId,
+      showMessageStatus
+    ]
   );
 
   const localeStrings = useLocale().strings.messageThread;
   const strings = useMemo(() => ({ ...localeStrings, ...props.strings }), [localeStrings, props.strings]);
 
   const defaultStatusRenderer: (
+    message: ChatMessage,
     status: MessageStatus,
-    messageThreadReadCount: number,
+    readReceiptForEachSender: { [key: string]: { lastReadMessage: string; name: string } },
```

Comment:
This type shows up for so many times, we can give it a type like message, maybe name it ReadReceiptsBySenderId

Reply:
I like `ReadReceiptsBySenderId`! I would make the property match also
``` suggestion
    readReceiptsBySenderId: ReadReceiptsBySenderId
```

Or perhaps just:
``` suggestion
    readReceipts: ReadReceiptsBySenderId
```